ensime_shared/editor.py

```diff
@@ -348,11 +346,8 @@ def cursorhold(self):
             class since it represents "the Vim plugin" as an entity, iterating
             through its list of clients and calling the logic for each.
         """
-        # Make sure no plugin overrides this
-        self._vim.options['updatetime'] = 1000
-        # Keys with no effect, just retrigger CursorHold
-        # http://vim.wikia.com/wiki/Timer_to_execute_commands_periodically
-        self._vim.command(r'call feedkeys("f\e")')
+        self._vim.eval("timer_start({}, 'EnRefreshMessages', {{'repeat': -1}})"
```

Comment:
This makes me think that each `Editor` instance shouldn't maintain its own timer as state. There could be a single timer reference as an attribute of `Ensime`. That's kind of the spirit of the todo comment just above this diff.

There can be more than one live instance of `Editor`—currently one is created for each `EnsimeClient` instance, meaning if you edit two ENSIME-enabled projects in one Vim session you'd end up with two timers. That could be changed (in fact #367 changes it, though I didn't think about that much), but I feel leery of having one `Editor` that needs to be treated as a singleton. As an abstraction it might make perfect sense that there _is_ one editor, but practically speaking singletons get ugly. Running with the extensions idea of #366, it'd be nice if third-party code can simply do `editor = Editor(vim)` for UI conveniences, instead of there being some global. It currently has virtually no significant state.

`Ensime` is already effectively a singleton, so it's nice to keep stuff of this nature there as long as it feels like a reasonable design and not overloaded, I think. Thoughts?